languages/python/sqlalchemy-oso/sqlalchemy_oso/roles2.py

```diff
@@ -764,6 +764,31 @@ def assignments_for_resource(self, resource, session=None):
         )
         return [{"user_id": ur.user_id, "role": ur.role} for ur in user_roles]
 
+    @ensure_configured
+    def assignments_for_user(self, user, session=None):
+        # List the role assignments for a specific user
+        if not session:
+            session = self._get_session()
+
+        user_pk_name = inspect(user.__class__).primary_key[0].name
+        user_id = getattr(user, user_pk_name)
```

Comment:
Are any of these lookups fallible? We needed some error handling for similar lookups elsewhere in the new roles code:

https://github.com/osohq/oso/commit/70bd20ca72308cdbdffb9d852062a8e1b65734ce

https://github.com/osohq/oso/commit/dbbb5a6d6a65f1b9eea82a528a1afde8bebb35e6